### PR TITLE
Update controller runtime version

### PR DESCRIPTION
**Pull request by pingsutw**
Signed-off-by: Kevin Su <pingsutw@apache.org>

# TL;DR
The version of controller runtime in the single binary is 0.12.1, but the one in admin is 0.8.3, causing compatibility issues.
Therefore, we should bump the controller runtime version in flyteadmin as well.

```
(base) ➜  flyte git:(flyte-bot-update-releases) docker build -t pingsutw/sandbox-lite-test -f docker/sandbox-lite/Dockerfile . --build-arg FLYTE_TEST=local
...
#32 102.2 github.com/flyteorg/flytepropeller/plugins
#32 124.4 # github.com/mattn/go-sqlite3
#32 124.4 sqlite3-binding.c: In function 'sqlite3SelectNew':
#32 124.4 sqlite3-binding.c:128049:10: warning: function may return address of local variable [-Wreturn-local-addr]
#32 124.4 128049 |   return pNew;
#32 124.4        |          ^~~~
#32 124.4 sqlite3-binding.c:128009:10: note: declared here
#32 124.4 128009 |   Select standin;
#32 124.4        |          ^~~~~~~
#32 124.4 gorm.io/driver/sqlite
#32 124.5 github.com/flyteorg/datacatalog/pkg/repositories
#32 124.5 github.com/flyteorg/flyteadmin/pkg/repositories
#32 124.6 github.com/flyteorg/datacatalog/pkg/manager/impl
#32 124.6 github.com/flyteorg/flyteadmin/pkg/clusterresource
#32 124.6 github.com/flyteorg/flyteadmin/scheduler
#32 124.6 github.com/flyteorg/flyteadmin/pkg/rpc/adminservice
#32 124.8 # github.com/flyteorg/flyteadmin/pkg/clusterresource
#32 124.8 /go/pkg/mod/github.com/flyteorg/flyteadmin@v1.1.33/pkg/clusterresource/utils.go:37:9: cannot use &strategicMergeFromPatch{…} (value of type *strategicMergeFromPatch) as type client.Patch in return statement:
#32 124.8 	*strategicMergeFromPatch does not implement client.Patch (wrong type for Data method)
#32 124.8 		have Data(obj "k8s.io/apimachinery/pkg/runtime".Object) ([]byte, error)
#32 124.8 		want Data(obj client.Object) ([]byte, error)
#32 124.8 github.com/flyteorg/datacatalog/pkg/rpc/datacatalogservice
#32 125.0 github.com/flyteorg/flyteadmin/pkg/server
------
executor failed running [/bin/sh -c go build -tags console -v -o /flyte cmd/main.go]: exit code: 2
```

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/pull/2773

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by katrogan** [comment]
can you test this with sandbox?

**Comment by pingsutw** [comment]
I've tested it with sandbox and demo. Both of them work.

**Review by katrogan — APPROVED** [review]
